### PR TITLE
Add Decontam

### PR DESCRIPTION
**Pull request by paulzierep**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/decontam/test-data/decontam_Rscript.Rmd
Comment (bgruening) [inline tools/decontam/test-data/decontam_Rscript.Rmd:1]: Is that file needed?

File: tools/decontam/decontam.xml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                <param name="control_metadata" value="control"/>
+            </conditional>
+            <param name="threshold" value="0.5"/>
+            <output name="decontam_phyloseq" file="phyloseq_output.rds" ftype="phyloseq"/>
+            <output name="decontam_otu" file="otu_output.tsv" ftype="tabular"/>
+            <output name="prevalence" file="Prevalence_Plot.png" ftype="png"/>
+            <output name="library_size_vs_control" file="Library_Size_vs_Control_Plot.png" ftype="png"/>
+        </test>
+        <test>
+        <conditional name="input_type">
```
Comment (bgruening) [inline tools/decontam/decontam.xml:124]: indentation

File: tools/decontam/test-data/decontam_Rscript.Rmd
Comment (paulzierep) [reply tools/decontam/test-data/decontam_Rscript.Rmd:1]: no, but it explains how the test data is generated, which I thought would be nice to have for later updates and such ... I took inspiration from: https://github.com/galaxyproject/tools-iuc/blob/09c6b3beb62716ed8175cb3c579d3c3b9ea847e9/tools/ampvis2/test-data/generate.R

**Comment by SaimMomin12** [comment]
The tests fails due to linting error in the `decontam/test-data/decontam_Rscript.Rmd`

This can be fixed by linting the R file using the [lintr R package](https://lintr.r-lib.org/).

**Comment by bgruening** [comment]
Mh, I think its better not to lint test-data at all.

File: tools/decontam/test-data/decontam_Rscript.Rmd
Comment (paulzierep) [reply tools/decontam/test-data/decontam_Rscript.Rmd:1]: so what do you think @bgruening keep it to make an update simpler (there is another function of this tool, for which the markdown would be good to have), or remove since not needed explicitly for the wrapper ?

File: tools/decontam/decontam.xml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    </configfiles>
+    <inputs>
+        <conditional name="input_type">
+            <param name="select_input" type="select" label="Phyloseq or Feature table input" help="This tool can work with phyloseq objects or feature table inputs.">
+                <option value="phyloseq">Phyloseq</option>
+                <option value="feature_table">Feature table</option>
+            </param>
+            <when value="phyloseq">
+                <param name="phyloseq_object" type="data" format="phyloseq" label="Phyloseq object"/>
+                <param name="control_metadata" type="text" label="Control column" help="Column in the phyloseq metadata specifying weather a samples is a negativ control (0 for normal samples / 1 for control)"/>
```
Comment (bgruening) [inline tools/decontam/decontam.xml:94]:
```suggestion
                <param name="control_metadata" type="text" label="Control column" help="Column in the phyloseq metadata specifying weather a sample is a negative control (0 for normal samples / 1 for control)"/>
```

File: tools/decontam/decontam.xml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            </when>
+            <when value="feature_table">
+                <param name="otu" type="data" format="tabular" label="Feature table" help="OTU/ASV or other feature table. The first column must have corresponding IDs to the metadata table."/>
+                <param name="metadata" type="data" format="tabular" label="Metadata" help="Metadata that contains a column specifying weather a samples is a negativ control (0 for normal samples / 1 for control). The first column must have corresponding IDs to the feature table."/>
+                <param name="control" type="data_column" data_ref="metadata" use_header_names="true" multiple="false" optional="false" label="Control column" help="Column specifying weather a samples is a negativ control (0 for normal samples / 1 for control)."/>
```
Comment (bgruening) [inline tools/decontam/decontam.xml:99]:
```suggestion
                <param name="control" type="data_column" data_ref="metadata" use_header_names="true" multiple="false" optional="false" label="Control column" help="Column specifying weather a sample is a negative control (0 for normal samples / 1 for control)."/>
```

File: tools/decontam/decontam.xml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    **Output**
+    If a phyloseq object is provided as input, the output phyloseq object is pruned to only contain non-contaminant features.
+    If feature table / metadata is provided as input, the output phyloseq object is pruned to only contain non-contaminant features and
+    does also not contain the TAX table. The output feature table is pruned to only contain non-contaminant features.
+
+    **Threshold**
+    The default threshold for a contaminant is that it reaches a probability of 0.1 in the statistical test being performed. 
+    In the prevalence test there is a special value worth knowing, threshold=0.5, that will identify as contaminants all 
+    sequences thare are more prevalent in negative controls than in positive samples. 
+    ]]>
```
Comment (bgruening) [inline tools/decontam/decontam.xml:154]:
```suggestion
    Simple statistical identification of contaminating sequence features in marker-gene or metagenomics data. 
    Works on any kind of feature derived from environmental sequencing data (e.g. ASVs, OTUs, taxonomic groups, MAGs,...). 
    Requires DNA quantitation data or sequenced negative control samples.

    Note: Currently only the negative control samples method is implemented in this wrapper.

    **Output**
    If a phyloseq object is provided as input, the output phyloseq object is pruned to only contain non-contaminant features.
    If the feature table/metadata is provided as input, the output phyloseq object is pruned only to contain non-contaminant features and
    not contain the TAX table. The output feature table is pruned only to contain non-contaminant features.

    **Threshold**
    The default threshold for a contaminant is that it reaches a probability of 0.1 in the statistical test being performed. 
    In the prevalence test there is a special value worth knowing, threshold=0.5, that will identify as contaminants all 
    sequences that are more prevalent in negative controls than in positive samples. 
    ]]>
```

**Review by bgruening — APPROVED** [review]
